cirq-core/cirq/qis/channels.py

```diff
@@ -75,6 +75,38 @@ def kraus_to_superoperator(kraus_operators: Sequence[np.ndarray]) -> np.ndarray:
     return m
 
 
+def superoperator_to_kraus(superoperator: np.ndarray) -> Sequence[np.ndarray]:
+    """Returns a Kraus representation of a channel specified via the superoperator matrix."""
+    return choi_to_kraus(superoperator_to_choi(superoperator))
+
+
+def choi_to_superoperator(choi: np.ndarray) -> np.ndarray:
+    """Returns the superoperator matrix of a quantum channel specified via the Choi matrix."""
+    d = int(np.round(np.sqrt(choi.shape[0])))
+    if choi.shape != (d * d, d * d):
+        raise ValueError(f"Invalid Choi matrix shape, expected {(d * d, d * d)}, got {choi.shape}")
+    if not np.allclose(choi, choi.T.conj()):
+        raise ValueError("Choi matrix must be Hermitian")
+
+    c = np.reshape(choi, (d, d, d, d))
+    s = np.swapaxes(c, 1, 2)
+    return np.reshape(s, (d * d, d * d))
+
+
+def superoperator_to_choi(superoperator: np.ndarray) -> np.ndarray:
+    """Returns the Choi matrix of a quantum channel specified via the superoperator matrix."""
```

Comment:
do we have documentation anywhere that defines what these representations are? If no: we should beef up these docstrings to give some context for the conversion.

In any case, it could be helpful to give a little note about how "involved" each conversion is. E.g. add a line: "Conversions between superoperator and choi representations only involve axes manipulation"

"Conversions to kraus require an eigendecomposition"

Reply:
Good point. I'll send thorough updated to docstrings in a separate PR.

Reply:
Done in #4554 (both definitions of the mathematical objects and the notes about complexity).